lib/fluent/plugin/prometheus.rb

```diff
@@ -88,19 +88,20 @@ def self.placeholder_expander(log)
 
       def configure(conf)
         super
+        @placeholder_values = {}
         @placeholder_expander = Fluent::Plugin::Prometheus.placeholder_expander(log)
         @hostname = Socket.gethostname
       end
 
       def instrument(tag, es, metrics)
-        placeholder_values = {
+        @placeholder_values[tag] ||= {
```

Comment:
How about assigning to local variable together?
It can skip `[]` access in `each`.

Reply:
How about `values = (@placeholder_values[tag] ||= {...})`?

Reply:
it looks too much for me(it's just IMO) because there is only one `[]` access in this method. 
I think it's decreasing readability. if there is an obvious performance degradation, it should be changed to like that.
But as long as it isn't a problem, the current code is fine for me.

Reply:
If you are ok, it's ok.
My concern is `@placeholder_values` are never clear so `[]` cost may become high on long running environment with lots of tag.

Reply:
I see. I ran benchmark. I think this difference is ignorable.

```rb
require 'benchmark/ips'

hash1 = {}
hash2 = {}

10000000.times do |i|
  hash1[i] = i
end

10.times do |i|
  hash2[i] = i
end

puts "Ruby: #{RUBY_VERSION}"

Benchmark.ips do |x|
  x.report('hash stored many keys') do
    hash[8]
  end

  x.report("hash stored") do
    hash[8]
  end
end

```

```
Ruby: 2.6.3
Warming up --------------------------------------
hash stored many keys
                       406.466k i/100ms
         hash stored   410.497k i/100ms
Calculating -------------------------------------
hash stored many keys
                         16.367M (± 3.3%) i/s -     82.106M in   5.021938s
         hash stored     16.655M (± 4.5%) i/s -     83.331M in   5.013647s
```

Reply:
looks good.